### ISSUE TITLE
addObject with CoffeeScript

### ISSUE THREAD

**Issue by neezer** [ryanflorence/jQuery.addObject#1]
Just as an FYI, when using your plugin pattern and adding via `addObject`, one should add a clean return to the end of the init, like this:

``` coffeescript
MyObject = (selector, options) ->

    this.jQuery = $ selector
    this.options = $.extend this.defaults, options

    return

MyObject.prototype = {

  defaults: {
    one: 1
    two: 2
  }

  foo: 'bar'

  # etc...

}
```

I ran into this "gotcha" earlier and thought I'd share. :)

Btw, this pattern is awesome!

**Comment by ryanflorence** [comment]
What's the issue?  When you use `new` it ignores the return of the constructor function, so it shouldn't matter.  Maybe a weird coffeescript thing?

**Comment by neezer** [comment]
The issue (so far as I can tell, and I'm no expert) is that CoffeeScript `return`s the last line in a function block, so if that function in turn returns an element instead of itself, then addObject assigns `this.data` to the wrong object. (In my case, it was returning the last created SVG element, rather than the whole object I initially passed in to the plugin).

I was writing a plugin using addObject and left the last line of my init block as

``` coffeescript
MyObject = (selector, options) ->

    @jQuery = $ selector
    @options = $.extend this.defaults, options

    @create()
```

So when I ran it, I got an error, which I tracked back to this line in your addObject plugin:

``` javascript
var prop = instance[arg];
```

... which was complaining because `instance` was undefined.

I only had this issue when trying to run methods on already instantiated instances of my plugin, like `$('#element').myNameSpace('foo');`. Adding that extra `return` at the end solved this issue for me.

Probably is a weird CoffeeScript thing and not anything wrong with your plugin; I just wanted to share my findings in case anyone else had that issue.
